tests/model/architecture.test.js

```diff
@@ -0,0 +1,190 @@
+            expect(a.ySpan).toBe(51);
+            expect(a.hasParams).toBe(false);
+        });
+
+        test('default', () => {
+            let a = new Architecture();
+
+            expect(a.name).toBe(Validation.DEFAULT_STR_VALUE);
+            expect(a.layers).toEqual([]);
+            expect(a.xSpan).toBe(Validation.DEFAULT_SPAN_VALUE);
+            expect(a.ySpan).toBe(Validation.DEFAULT_SPAN_VALUE);
+            expect(a.hasParams).toBe(false);
+        });
+    });
+
+    test('modify fields', () => {
+        let a = new Architecture();
+
+        a.name = 'arch';
+        a.layers = [topLayer, botLayer];
+        a.xSpan = 10;
+        a.ySpan = 15;
+        a.hasParams = true;
+
+        expect(a.name).toBe('arch');
+        expect(a.layers).toEqual([topLayer, botLayer]);
+        expect(a.xSpan).toBe(10);
+        expect(a.ySpan).toBe(15);
+        expect(a.hasParams).toBe(true);
+    });
+});
+
+describe('determineParams', () => {
+    test('does have params', () => {
+        let a = new Architecture();
+
+        a.xSpan = 10;
+        a.ySpan = 15;
+        expect(a.determineParams()).toBe(true);
+
+        expect(a.hasParams).toBe(true);
+    });
+
+    test('does not have params', () => {
+        let a = new Architecture();
+
+        a.xSpan = -14;
+        a.ySpan = -9876543;
+        expect(a.determineParams()).toBe(false);
+
+        expect(a.hasParams).toBe(false);
+    });
+});
+
+describe('validation', () => {
+    describe('valid', () => {
+        test('with params', () => {
+            let a = new Architecture('arch', [topLayer, botLayer], 1000, 1000);
+
+            expect(a.validate()).toBe(true);
+        });
+
+        test('without params', () => {
+            let a = new Architecture('arch', [topLayer, botLayer]);
+
+            expect(a.validate()).toBe(true);
+        });
+    });
+
+    describe('invalid', () => {
+        test('all fields', () => {
+            let a = new Architecture();
+
+            expect(a.validate()).toBe(false);
+        });
+
+        test('name', () => {
+            let a = new Architecture(Validation.DEFAULT_STR_VALUE, [topLayer, botLayer]);
+
+            expect(a.validate()).toBe(false);
+        });
+
+        describe('layers', () => {
+            test('empty', () => {
+                let a = new Architecture('arch', []);
+
+                expect(a.validate()).toBe(false);
+            });
+
+            test('invalid component', () => {
+                let a = new Architecture('arch', [new Layer('bad-layer', 'unique-bad-layer-id', [new Component()],
+                        [connection1]), botLayer]);
+
+                expect(a.validate()).toBe(false);
+            });
+
+            test('invalid connection', () => {
+                let a = new Architecture('arch', [topLayer, new Layer('bad-layer', 'unique-bad-layer-id', [component1],
+                        [new Connection()])]);
+
+                expect(a.validate()).toBe(false);
```

Comment:
This isn't bad, but it could be considered overkill over over-coverage since you probably test for these types of failure cases when you test for valid/invalid layers. The only issue with this is it can make failures noisy because they seem like they are coming in at a lot of places but really its just the same type of test in multiple different locations.

Reply:
I was thinking that it would be a good idea to test everything, but you're right. Should it be removed?

Reply:
You can open another PR to look at coverage and hash out what should go where since this might be something we should look at across a number of test suites.